src/GrantType/GrantTypeBase.php

```diff
@@ -87,8 +150,28 @@ public function getToken()
         }
 
         $response = $this->client->post($config['token_url'], $requestOptions);
-        $data = $response->json();
 
-        return new AccessToken($data['access_token'], $data['token_type'], $data);
+        return $response->json();
+    }
+
+    /**
+     * compute the current token cache key
+     *
+     * @param $config
+     *
+     * @return string
+     */
+    protected function getCacheKey($config)
```

Comment:
You don't need to pass $config around as a variable: there is already `$this->config`